src/tpm.rs

```diff
@@ -0,0 +1,220 @@
+use volatile_register::{RW};
+use bit_field::BitField;
+use port::PwmPin;
+use sim::ClockGate;
+
+//KL26 manual - pp 570-571
+const TPM0_ADDR: usize = 0x4003_8000;
+const TPM1_ADDR: usize = 0x4003_9000;
+const TPM2_ADDR: usize = 0x4003_A000;
+
+#[derive(Clone, Copy)]
+pub enum TimerNum {
+    TPM0,
+    TPM1,
+    TPM2
+}
+
+#[repr(C,packed)]
+struct TPMRegs {
+    sc:         RW<u32>,
+    cnt:        RW<u32>,
+    modu:       RW<u32>,
```

Comment:
Thought, name this `mod_` as per the Rust formatting guidelines. https://github.com/rust-lang-nursery/fmt-rfcs/blob/master/guide/advice.md#names